pkg/image/airgap.go

```diff
@@ -675,6 +700,107 @@ func reportWriterWithProgress(imageInfos map[string]*imagetypes.ImageInfo, repor
 	return pipeWriter
 }
 
+type OCIArtifactFile struct {
+	Name      string
+	Path      string
+	MediaType string
+}
+
+func PushEmbeddedClusterArtifacts(airgapBundle string, opts imagetypes.PushEmbeddedArtifactsOptions) error {
+	embeddedClusterDir := filepath.Join(airgapBundle, "embedded-cluster")
+	if _, err := os.Stat(embeddedClusterDir); err != nil {
```

Comment:
how about also checking if we're actually running in an embedded cluster? might be a good guardrail to have.